plugin/evm/limit_order.go

```diff
@@ -59,10 +61,30 @@ func NewLimitOrderProcesser(ctx *snow.Context, chainConfig *params.ChainConfig,
 		backend:      backend,
 		memoryDb:     limitorders.NewInMemoryDatabase(),
 		orderBookABI: orderBookAbi,
+		blockChain: blockChain,
 	}
 }
 
 func (lop *limitOrderProcesser) ListenAndProcessTransactions() {
+	lastAccepted := lop.blockChain.LastAcceptedBlock().NumberU64()
+	if lastAccepted > 0 {
+		log.Info("ListenAndProcessTransactions - beginning sync", " till block number", lastAccepted)
+		
+		allTxs := types.Transactions{}
+		for i := uint64(0); i <= lastAccepted; i++ {
+			block := lop.blockChain.GetBlockByNumber(i)
+			if block != nil {
+				allTxs = append(allTxs, block.Transactions()...)
+			}
+		}
+	
+		for _, tx := range allTxs {
+			parseTx(lop.orderBookABI, lop.memoryDb, *lop.backend, tx)
+		}
+
+		log.Info("ListenAndProcessTransactions - sync complete", "till block number", lastAccepted, "total transactions", len(allTxs))
+	}
+
```

Comment:
instead of iterating through all blocks again, do you think it might be a good idea to write a `lop` hook at the place where this already happens? - i.e. new blocks are fetched and added to state.